scipy/stats/_unuran/unuran_callback.h

```diff
@@ -0,0 +1,271 @@
+#include <Python.h>
+#include <setjmp.h>
+#include "ccallback.h"
+#include "unuran.h"
+
+/*
+ * A common UNU.RAN thunk.
+ * 
+ * Parameters
+ * ----------
+ * CAST_FUNC :
+ *     CPython function to convert the C data type of the input i.e. data type of
+ *     the quantile (`ARG`) into a PyObject.
+ * CAST_BACK_FUNC :
+ *     CPython function to convert the PyObject result to a C data type. i.e the
+ *     return type of the thunk.
+ * FUNCNAME :
+ *     The name of the function to call in the thunk. This name must be present in
+ *     the `callbacks` dictionary.
+ * ARG :
+ *     The argument to treat as an input to the Python function.
+ * 
+ * Returns
+ * -------
+ * result :
+ *     The value returned by the Python function after converting it into a C type
+ *     using `CAST_BACK_FUNC`.
+ */
+#define UNURAN_THUNK(CAST_FUNC, CAST_BACK_FUNC, FUNCNAME, ARG)                         \
+    ccallback_t *err_callback = ccallback_obtain();                                    \
+    unuran_callback_t *unur_callback = (unuran_callback_t *)(err_callback->info_p);    \
+    PyObject *extra_arguments = unur_callback->params;                                 \
+    PyObject *py_function = PyDict_GetItemString(unur_callback->callbacks, FUNCNAME);  \
```

Comment:
This is where I obtain the `ccallback_t` object and extract the dictionary of callbacks where Python callables are present. The required function will be present at key `FUNCNAME` (which could be `"pdf"`, `"dpdf"`, etc). This way I can access them all at the same time without writing a separate thunk and creating a separate callback.